### ISSUE TITLE
AmqpTcpEndpoint.ToString() does not report a Uri scheme of "amqps" although SslOption is enabled

### ISSUE THREAD

**Issue by djoe47441** [rabbitmq/rabbitmq-dotnet-client#861]
`AmqpTcpEndpoint.ToString()` should correctly report a Uri scheme of _amqps_ whenever its SSLOption is enabled.

From [AmqpTcpEndpoint.cs, line 282 ff.](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/aa5e8cb17edb3a830a65be7e2a0cfbe7029c6605/projects/RabbitMQ.Client/client/api/AmqpTcpEndpoint.cs#L282):

```
        /// <summary>
        /// Returns a URI-like string of the form amqp-PROTOCOL://HOSTNAME:PORTNUMBER.
        /// </summary>
        /// <remarks>
        /// This method is intended mainly for debugging and logging use.
        /// </remarks>
        public override string ToString()
        {
            return $"amqp://{HostName}:{Port}";
        }
```

Compare this to the implementation of `ConnectionFactory.SetUri(...)` [ConnectionFactory.cs, line 531 ff.](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/aa5e8cb17edb3a830a65be7e2a0cfbe7029c6605/projects/RabbitMQ.Client/client/api/ConnectionFactory.cs#L531) where an Uri scheme of _amqps_ is correctly interpreted to initialize the SSLOption:

```
        private void SetUri(Uri uri)
        {
            Endpoint = new AmqpTcpEndpoint();

            if (string.Equals("amqp", uri.Scheme, StringComparison.OrdinalIgnoreCase))
            {
                // nothing special to do
            }
            else if (string.Equals("amqps", uri.Scheme, StringComparison.OrdinalIgnoreCase))
            {
                Ssl.Enabled = true;
                Ssl.Version = AmqpUriSslProtocols;
                Ssl.AcceptablePolicyErrors = SslPolicyErrors.RemoteCertificateNameMismatch;
                Port = AmqpTcpEndpoint.DefaultAmqpSslPort;
            }
            else
            {
                throw new ArgumentException($"Wrong scheme in AMQP URI: {uri.Scheme}");
            }
```

**Comment by michaelklishin** [comment]
This is open source software. If you understand what exactly should be changed and why, please submit a pull request and it will be considered.

**Comment by lukebakken** [comment]
Fixed by #864
